drivers/video/video_mcux_csi.c

```diff
@@ -116,25 +117,62 @@ static void __frame_done_cb(CSI_Type *base, csi_handle_t *handle,
 	return;
 }
 
-static int video_mcux_csi_set_fmt(const struct device *dev,
-				  enum video_endpoint_id ep,
+#if defined(CONFIG_VIDEO_MCUX_MIPI_CSI2RX)
+K_HEAP_DEFINE(csi_heap, 1000);
+static struct video_format_cap *fmts;
+/*
+ * On i.MX RT11xx SoCs which have MIPI CSI-2 Rx, image data from the camera sensor after passing
+ * through the pipeline (MIPI CSI-2 Rx --> Video Mux --> CSI) will be implicitly converted to a
+ * 32-bits pixel format. For example, an input in RGB565 or YUYV (2-bytes format) will become a
+ * XRGB32 or XYUV32 (4-bytes format) respectively, at the output of the CSI.
+ */
+static inline void video_pix_fmt_convert(struct video_format *fmt, bool isGetFmt)
+{
+	switch (fmt->pixelformat) {
+	case VIDEO_PIX_FMT_XRGB32:
+		fmt->pixelformat = isGetFmt ? VIDEO_PIX_FMT_XRGB32 : VIDEO_PIX_FMT_RGB565;
+		break;
+	case VIDEO_PIX_FMT_XYUV32:
+		fmt->pixelformat = isGetFmt ? VIDEO_PIX_FMT_XYUV32 : VIDEO_PIX_FMT_YUYV;
+		break;
+	case VIDEO_PIX_FMT_RGB565:
+		fmt->pixelformat = isGetFmt ? VIDEO_PIX_FMT_XRGB32 : VIDEO_PIX_FMT_RGB565;
+		break;
+	case VIDEO_PIX_FMT_YUYV:
+		fmt->pixelformat = isGetFmt ? VIDEO_PIX_FMT_XYUV32 : VIDEO_PIX_FMT_YUYV;
+		break;
+	}
+
+	fmt->pitch = fmt->width * video_pix_fmt_bpp(fmt->pixelformat);
+}
+#endif
+
+static int video_mcux_csi_set_fmt(const struct device *dev, enum video_endpoint_id ep,
 				  struct video_format *fmt)
 {
 	const struct video_mcux_csi_config *config = dev->config;
 	struct video_mcux_csi_data *data = dev->data;
 	unsigned int bpp = video_pix_fmt_bpp(fmt->pixelformat);
 	status_t ret;
+	struct video_format format = *fmt;
 
 	if (!bpp || ep != VIDEO_EP_OUT) {
 		return -EINVAL;
 	}
 
-	data->pixelformat = fmt->pixelformat;
```

Comment:
This can be deleted from `video_mcux_csi_data`

Reply:
Yes, it is addressed and merged.